less-openui5 0.3.0+ unit tests fail on Windows

This issue originates from https://github.com/SAP/openui5/issues/1440 but is transferred here as the proper repository.

Issue details: 

less-openui5 version:  0.3.1 (starting in 0.3.0)

Summary:  less-openui5 unit tests fail on Windows due to mishandling of backslashes.

Steps to reproduce the problem:
(using Node v6.9.1, NPM v3.10.8, git version 2.11.1.windows.prerelease.2)
1.  Open up a Git Bash shell on Windows.
2.  In an empty directory, run git clone https://github.com/SAP/less-openui5.git
3.  Change the directory to less-openui5.
4.  Run npm install.
5.  Run npm test.

What is the expected result? The tests pass with no issues. (They do with less-openui v0.2.0.)

What happens instead? 
> less-openui5@0.3.1 test C:\temp\less-openui5
> npm run lint && npm run unit

> less-openui5@0.3.1 lint C:\temp\less-openui5
> eslint lib test

... (Lint warnings removed) ...

> less-openui5@0.3.1 unit C:\temp\less-openui5
> mocha test/*.js

  performance workaround
    1) should run with patched String prototype

  options
    √ should return css, cssRtl, variables and imports with default options (lessInput)
    √ should return css, cssRtl, variables and imports with default options (lessInputPath)
    √ should not return cssRtl with option rtl set to false
    √ should return minified css and cssRtl with lessOption compress set to true
    √ should resolve import directives with rootPaths option
    √ should read input file with rootPaths option

  libraries
    2) should create base theme
    3) should create foo theme with scope as defined in .theming file
    4) should create bar theme with scope as defined in .theming file

  error handling
    √ should have correct error in case of undefined variable usage

  rtl
    √ background-position
    √ background
    √ border
    √ cursor
    √ gradient
    √ image-url
    √ misc
    √ shadow
    √ transform
    √ variables

  variables
    √ should return only globally defined variables

  imports
    √ should return imported file paths
    √ should use "relativeUrls" parser option by default
    √ should not rewrite urls when "relativeUrls" parser option is set to "false"

  inline theming parameters
    √ should not include inline parameters when no library name is given
    √ should include inline parameters when library name is given

  theme caching
    √ should cache the theme
    √ should recompile the theme after clearing the cache

  CSS Scoping of
    comments
      √ should return same CSS for foo
      √ should return same CSS for bar
    css-scope-root
      √ should return same CSS for foo
      √ should return same CSS for bar
    default
      √ should return same CSS for foo
      √ should return same CSS for bar
    dom
      √ should return same CSS for foo
      √ should return same CSS for bar
    empty media queries
      √ should return same CSS for foo
      √ should return same CSS for bar
    HTML
      √ should return same CSS for foo
      √ should return same CSS for bar
    media-queries
      √ should return same CSS for foo
      √ should return same CSS for bar
    mixins
      √ should return same CSS for foo
      √ should return same CSS for bar
    multiple imports
      √ should return same CSS for foo
      √ should return same CSS for bar

  43 passing (321ms)
  4 failing

  1) performance workaround should run with patched String prototype:

      AssertionError: import list should be correct.
      + expected - actual

       [
      -  "test\\fixtures\\libraries\\lib1\\my\\ui\\lib\\themes\\foo\\library.source.less"
      -  "test\\fixtures\\libraries\\lib1\\my\\ui\\lib\\themes\\base\\library.source.less"
      -  "test\\fixtures\\libraries\\lib2\\my\\ui\\lib\\themes\\bar\\library.source.less"
      -  "test\\fixtures\\libraries\\lib1\\sap\\ui\\core\\themes\\foo\\.theming"
      +  "test/fixtures/libraries/lib1/my/ui/lib/themes/foo/library.source.less"
      +  "test/fixtures/libraries/lib1/my/ui/lib/themes/base/library.source.less"
      +  "test/fixtures/libraries/lib2/my/ui/lib/themes/bar/library.source.less"
      +  "test/fixtures/libraries/lib1/sap/ui/core/themes/foo/.theming"
       ]

      at test\test-perf-workaround.js:64:14

  2) libraries should create base theme:

      AssertionError: import list should be correct.
      + expected - actual

       [
      -  "test\\fixtures\\libraries\\lib1\\my\\ui\\lib\\themes\\base\\library.source.less"
      +  "test/fixtures/libraries/lib1/my/ui/lib/themes/base/library.source.less"
       ]

      at test\test.js:164:14

  3) libraries should create foo theme with scope as defined in .theming file:

      AssertionError: import list should be correct.
      + expected - actual

       [
      -  "test\\fixtures\\libraries\\lib1\\my\\ui\\lib\\themes\\foo\\library.source.less"
      -  "test\\fixtures\\libraries\\lib1\\my\\ui\\lib\\themes\\base\\library.source.less"
      -  "test\\fixtures\\libraries\\lib2\\my\\ui\\lib\\themes\\bar\\library.source.less"
      -  "test\\fixtures\\libraries\\lib1\\sap\\ui\\core\\themes\\foo\\.theming"
      +  "test/fixtures/libraries/lib1/my/ui/lib/themes/foo/library.source.less"
      +  "test/fixtures/libraries/lib1/my/ui/lib/themes/base/library.source.less"
      +  "test/fixtures/libraries/lib2/my/ui/lib/themes/bar/library.source.less"
      +  "test/fixtures/libraries/lib1/sap/ui/core/themes/foo/.theming"
       ]

      at test\test.js:200:14

  4) libraries should create bar theme with scope as defined in .theming file:

      AssertionError: import list should be correct.
      + expected - actual

       [
      -  "test\\fixtures\\libraries\\lib2\\my\\ui\\lib\\themes\\bar\\library.source.less"
      -  "test\\fixtures\\libraries\\lib1\\my\\ui\\lib\\themes\\foo\\library.source.less"
      -  "test\\fixtures\\libraries\\lib1\\my\\ui\\lib\\themes\\base\\library.source.less"
      -  "test\\fixtures\\libraries\\lib2\\sap\\ui\\core\\themes\\bar\\.theming"
      +  "test/fixtures/libraries/lib2/my/ui/lib/themes/bar/library.source.less"
      +  "test/fixtures/libraries/lib1/my/ui/lib/themes/foo/library.source.less"
      +  "test/fixtures/libraries/lib1/my/ui/lib/themes/base/library.source.less"
      +  "test/fixtures/libraries/lib2/sap/ui/core/themes/bar/.theming"
       ]

      at test\test.js:238:14

--- a/test/test-perf-workaround.js
+++ b/test/test-perf-workaround.js
@@ -16,6 +16,7 @@
 'use strict';
 
 var assert = require('assert');
+var path = require('path');
 var readFile = require('./common/helper').readFile;
 
 // tested module
@@ -62,10 +63,10 @@ describe('performance workaround', function() {
       assert.equal(result.cssRtl, readFile('test/expected/libraries/lib1/my/ui/lib/themes/foo/library-RTL.css'), 'rtl css should be correctly generated.');
       assert.deepEqual(result.variables, oVariablesExpected, 'variables should be correctly collected.');
       assert.deepEqual(result.imports, [
-        "test/fixtures/libraries/lib1/my/ui/lib/themes/foo/library.source.less",
-        "test/fixtures/libraries/lib1/my/ui/lib/themes/base/library.source.less",
-        "test/fixtures/libraries/lib2/my/ui/lib/themes/bar/library.source.less",
-        "test/fixtures/libraries/lib1/sap/ui/core/themes/foo/.theming"
+        path.join("test", "fixtures", "libraries", "lib1", "my", "ui", "lib", "themes", "foo", "library.source.less"),
+        path.join("test", "fixtures", "libraries", "lib1", "my", "ui", "lib", "themes", "base", "library.source.less"),
+        path.join("test", "fixtures", "libraries", "lib2", "my", "ui", "lib", "themes", "bar", "library.source.less"),
+        path.join("test", "fixtures", "libraries", "lib1", "sap", "ui", "core", "themes", "foo", ".theming")
       ], 'import list should be correct.');
 
       assert.strictEqual(String.prototype.__lookupGetter__("customGetter"), customGetter, "Custom getter should again be set on String prototype.");

--- a/test/test.js
+++ b/test/test.js
@@ -162,7 +162,7 @@ describe('libraries', function() {
       assert.equal(result.cssRtl, readFile('test/expected/libraries/lib1/my/ui/lib/themes/base/library-RTL.css'), 'rtl css should be correctly generated.');
       assert.deepEqual(result.variables, { color1: '#fefefe' }, 'variables should be correctly collected.');
       assert.deepEqual(result.imports, [
-        "test/fixtures/libraries/lib1/my/ui/lib/themes/base/library.source.less"
+        path.join("test", "fixtures", "libraries", "lib1", "my", "ui", "lib", "themes", "base", "library.source.less")
       ], 'import list should be correct.');
 
     });
@@ -198,10 +198,10 @@ describe('libraries', function() {
       assert.equal(result.cssRtl, readFile('test/expected/libraries/lib1/my/ui/lib/themes/foo/library-RTL.css'), 'rtl css should be correctly generated.');
       assert.deepEqual(result.variables, oVariablesExpected, 'variables should be correctly collected.');
       assert.deepEqual(result.imports, [
-        "test/fixtures/libraries/lib1/my/ui/lib/themes/foo/library.source.less",
-        "test/fixtures/libraries/lib1/my/ui/lib/themes/base/library.source.less",
-        "test/fixtures/libraries/lib2/my/ui/lib/themes/bar/library.source.less",
-        "test/fixtures/libraries/lib1/sap/ui/core/themes/foo/.theming"
+        path.join("test", "fixtures", "libraries", "lib1", "my", "ui", "lib", "themes", "foo", "library.source.less"),
+        path.join("test", "fixtures", "libraries", "lib1", "my", "ui", "lib", "themes", "base", "library.source.less"),
+        path.join("test", "fixtures", "libraries", "lib2", "my", "ui", "lib", "themes", "bar", "library.source.less"),
+        path.join("test", "fixtures", "libraries", "lib1", "sap", "ui", "core", "themes", "foo", ".theming")
       ], 'import list should be correct.');
 
     });
@@ -236,10 +236,10 @@ describe('libraries', function() {
       assert.equal(result.cssRtl, readFile('test/expected/libraries/lib2/my/ui/lib/themes/bar/library-RTL.css'), 'rtl css should be correctly generated.');
       assert.deepEqual(result.variables, oVariablesExpected, 'variables should be correctly collected.');
       assert.deepEqual(result.imports, [
-        "test/fixtures/libraries/lib2/my/ui/lib/themes/bar/library.source.less",
-        "test/fixtures/libraries/lib1/my/ui/lib/themes/foo/library.source.less",
-        "test/fixtures/libraries/lib1/my/ui/lib/themes/base/library.source.less",
-        "test/fixtures/libraries/lib2/sap/ui/core/themes/bar/.theming"
+        path.join("test", "fixtures", "libraries", "lib2", "my", "ui", "lib", "themes", "bar", "library.source.less"),
+        path.join("test", "fixtures", "libraries", "lib1", "my", "ui", "lib", "themes", "foo", "library.source.less"),
+        path.join("test", "fixtures", "libraries", "lib1", "my", "ui", "lib", "themes", "base", "library.source.less"),
+        path.join("test", "fixtures", "libraries", "lib2", "sap", "ui", "core", "themes", "bar", ".theming")
       ], 'import list should be correct.');
 
     });

